suite/contracts/contracts_volume_test.go

```diff
@@ -0,0 +1,155 @@
+		for _, n := range d.Nodes {
+			fluxSpec := &client.FluxMonitorJobSpec{
+				Name:              fmt.Sprintf("%s_%d", d.Spec.JobPrefix, d.Index),
+				ContractAddress:   fluxInstance.Address(),
+				PollTimerPeriod:   d.Spec.NodePollTimePeriod,
+				PollTimerDisabled: false,
+				ObservationSource: client.ObservationSourceSpec(d.Adapter.InsideDockerAddr + "/variable"),
+			}
+			_, err = n.CreateJob(fluxSpec)
+			Expect(err).ShouldNot(HaveOccurred())
+		}
+		d.FluxInstancesMu.Lock()
+		d.FluxInstances = append(d.FluxInstances, fluxInstance)
+		d.FluxInstancesMu.Unlock()
+	}()
+}
+
+// DeployVolumeFlux deploys AggregatorsNum aggregators, creates jobs on all nodes for every aggregator
+func DeployVolumeFlux(spec *VolumeSpec) (*suite.DefaultSuiteSetup, []contracts.FluxAggregator, tools.ExternalAdapter) {
+	s, err := suite.DefaultLocalSetup(spec.InitFunc)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	clNodes, nodeAddrs, err := suite.ConnectToTemplateNodes()
+	// TODO: just keep 3 before we have k8s
+	oraclesAtTest := nodeAddrs[:3]
+	clNodesAtTest := clNodes[:3]
+	Expect(err).ShouldNot(HaveOccurred())
+	err = suite.FundTemplateNodes(s.Client, s.Wallets, clNodes, 2e18, 0)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	s.Client.(*client.EthereumClient).BorrowedNonces(true)
+
+	adapter := tools.NewExternalAdapter()
+
+	var fluxInstances []contracts.FluxAggregator
+	mu := &sync.Mutex{}
+	wg := &sync.WaitGroup{}
+	for i := 0; i < spec.AggregatorsNum; i++ {
+		wg.Add(1)
+		deployInstance(&InstanceDeployment{
+			Wg:              wg,
+			Index:           i,
+			Suite:           s,
+			Spec:            spec,
+			Oracles:         oraclesAtTest,
+			Nodes:           clNodesAtTest,
+			Adapter:         adapter,
+			FluxInstancesMu: mu,
+			FluxInstances:   fluxInstances,
+		})
+	}
+	wg.Wait()
+	return s, fluxInstances, adapter
+}
+
+var _ = FDescribe("Volume tests", func() {
+	var iteration = 1
+	jobPrefix := "flux_monitor"
+	s, _, adapter := DeployVolumeFlux(&VolumeSpec{
+		AggregatorsNum:     10,
+		JobPrefix:          jobPrefix,
+		NodePollTimePeriod: 15 * time.Second,
+		InitFunc:           client.NewHardhatNetwork,
+		FluxOptions:        contracts.DefaultFluxAggregatorOptions(),
+	})
+	prom := tools.NewPrometheusClient(s.Config.Prometheus)
+	Measure("Should process one round with 100 contracts, 100 submits per node", func(b Benchmarker) {
+		// all contracts/nodes/jobs are setup at that point, triggering new round,
+		// waiting for all contracts to complete one round and get metrics
+		err := adapter.TriggerValueChange(iteration)
+		if err != nil {
+			log.Fatal().Err(err)
+		}
+		// TODO: check all contracts have one round completed, options are:
+		// TODO: 1. poll prom metrics for last seen round, ensure all nodes have complete next round and there is no errors
+		// TODO: 2. override contract submit method, when round is complete call another collector contract
+		// TODO: when all nodes have complete one round listen to RoundComplete event
```

Comment:
For #2 I mean that metrics:
```
# HELP flux_monitor_reported_round Flux monitor's last reported round
# TYPE flux_monitor_reported_round gauge
flux_monitor_reported_round{job_spec_id="1"} 10
flux_monitor_reported_round{job_spec_id="10"} 12
flux_monitor_reported_round{job_spec_id="2"} 9
```

Reply:
For #2 we can override submit method and measure time from round start to round end, also call another contract (volume_collector. sol) when submissions are enough, collector emits an event when all nodes have complete one round

Reply:
Anyway, I did both, gonna check for prom metrics and ContractData across all deployed aggregators.